src/js/providers/utils/play-promise.ts

```diff
@@ -1,41 +1,34 @@
-export default function createPlayPromise(video) {
-    return new Promise(function(resolve, reject) {
+export default function createPlayPromise(video: HTMLVideoElement): Promise<void> {
+    return new Promise(function(resolve: () => void, reject: (reason: DOMException) => void): void {
         if (video.paused) {
-            return reject(playPromiseError('NotAllowedError', 0, 'play() failed.'));
+            return reject(new DOMException('play() failed.', 'NotAllowedError'));
```

Comment:
I don’t know if it’s safe to use DOMException as a constructor in all browsers. The recommendation was to use it as an interface

Reply:
Ah. I see now. Misread that comment. Whoops. 

Issues I run into with using `as DOMException` is that the `code` and `message` are readonly properties that I cannot assign, so I can't do:

> const error = new `Error`(message) as `DOMException`

But `Error` also does not contain the `code` property so I cannot do:

> return `error` as `DOMException`

Why I had originally made that new error type. As I look though, could potentially leverage `PlayerError` as I think Joe had mentioned...